### PR TITLE
Feature/rework documentation

### PR DESCRIPTION
**Pull request by FranFin**

### CODE REVIEW

File: docs/agent_api.rst [hunk 1]
```diff
@@ -0,0 +1,6 @@
+.. _agent_api_label:
+
+Agent API
+=========
+
+.. TODO: @José :))
```
Comment (FranFin) [inline docs/agent_api.rst:6]: @jamoralp All yours.

File: docs/getting_started.rst [hunk 2]
```diff
@@ -340,3 +406,85 @@ After this, we can close the transport used by the session.
...
+Publisher
+---------
+    .. image:: images/deployment_2.svg
+        :width: 600 px
+        :align: center
+
+Then, the `publisher client` is connected to the *Agent* `A`.
+This *Client* logs in a session with its *Client* key (0xAABBCCDD).
+
+.. TODO: check the above sentence: "logs in a session". And in general the use of "log" and "login".
```
Comment (FranFin) [inline docs/getting_started.rst:464]: @pablogs9 @jamoralp

File: docs/p2p.rst [hunk 3]
```diff
@@ -4,290 +4,83 @@ P2P Communication
...
+
+At this point, it's possible to launch the :code:`PublishHelloWorldClientP2P` executable
+located in the folder :code:`Micro-XRCE-DDS-Client/build/examples/PublishHelloWorldP2P`, which'll make
+the *Client* publish in the DDS World the :code:`HelloWorld` topic
+(take a look at the IDL defining this topic in the file
+:code:`Micro-XRCE-DDS-Client/examples/PublishHelloWorldP2P/HelloWorld.idl`). ::
+
+    $ Micro-XRCE-DDS-Client/build/examples/PublishHelloWorld2P2/PublishHelloWorldClientP2P 127.0.0.1 2019
+
+.. TODO: decide which path to give to the user.
```
Comment (FranFin) [inline docs/p2p.rst:67]: @pablogs9 Let's agree on the path.

File: docs/transport.rst [hunk 4]
```diff
@@ -227,6 +232,8 @@ In the best case, it is **only 19 bytes**, but it should be noted that, in this
 Custom Transport API
 --------------------
 
+.. TODO: This content is repeated in: client, client API. agent (not yet). agent API (not yet). What to do?
```
Comment (FranFin) [inline docs/transport.rst:235]: @pablogs9 @jamoralp

File: docs/transport.rst [hunk 5]
```diff
@@ -276,7 +283,7 @@ Where ``struct uxrCustomTransport*`` refers to the opened transport structure,
 
 Where ``struct uxrCustomTransport*`` refers to the opened transport structure,  the first ``uint8_t*`` is the buffer to be write with the received bytes, following ``size_t`` is the length of the buffer, the following ``int`` is the maximum time in milliseconds that the read operating should take and the last ``uint8_t*`` is an error code that should be set when the read process has any problem. This function should return the number of bytes received successfully.
 
-Server custom transport API
-^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Agent custom transport API
+^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-[JOSE]
+.. TODO: @Jose
```
Comment (FranFin) [inline docs/transport.rst:289]: @jamoralp

File: docs/agent.rst [hunk 6]
```diff
@@ -1,164 +1,238 @@
...
+
+.. _agent_cli:
+
+Agent CLI
+---------
+
+To run the *Agent*, first of all build it as indicated in the :ref:`installation_label` page.
+Once it is built successfully, launch it by executing one of the following commands:
+
+.. TODO: @jose: check and reorganize this.
```
Comment (FranFin) [inline docs/agent.rst:37]: @jamoralp

File: docs/agent.rst [hunk 7]
```diff
@@ -1,164 +1,238 @@
...
-      -r,--refs FILE                               Load a reference file
-      -v,--verbose UINT in {0,1,2,3,4,5,6}=4       Select log level from less to more verbose
-      -d,--discovery                               Activate the Discovery server
-      --disport UINT=7400 Needs: --discovery       Select the port for the Discovery server
-      --p2p UINT                                   Activate the P2P profile using the given port
+Custom transport
+----------------
 
-**Pseudo-Serial server** (only Linux) ::
+.. TODO: Add text here.
```
Comment (FranFin) [inline docs/agent.rst:120]: @jamoralp

File: docs/client.rst [hunk 8]
```diff
@@ -1,1151 +1,256 @@
...
+UDP           X           X        
+TCP           X           X        
+Serial        X                  
+Custom        X           X
+============ ========== =========
 
-.. code-block:: c
-    bool uxr_prepare_output_stream_fragmented(uxrSession* session, uxrStreamId stream_id, uxrObjectId datawriter_id,
-                                  struct ucdrBuffer* ub, size_t topic_size, uxrOnBuffersFull flush_callback);
+.. TODO: organize text below @pablo
```
Comment (FranFin) [inline docs/client.rst:138]: @pablogs9

File: docs/client_api.rst [hunk 9]
```diff
@@ -0,0 +1,1130 @@
...
+.. _transport_api:
+
+Transport
+^^^^^^^^^
+
+These functions are platform dependent.
+The declaration of these functions can be found in the ``uxr/client/profile/transport/`` folder.
+The common init transport functions follow the nomenclature below.
+
+.. TODO: new functions have been added. Should the description above be improved and expanded?
```
Comment (FranFin) [inline docs/client_api.rst:1011]: @pablogs9 @jamoralp

File: docs/quickstart.rst [hunk 10]
```diff
@@ -2,335 +2,227 @@
...
+    $ sudo make install
+
+After having installed the *Agent* system-wide, you can launch it.
+
+For this example, the *Client*-*Agent* communication will be done through UDP, using the port :code:`2019`
+and with the XML creation mode, which is the default mode for creating entities: ::
+
+    $ cd /usr/local/bin && MicroXRCEAgent udp -p 2019
+
+.. TODO: write a comment tagging guys and asking if it was ok to remove the "-r <references-file>" part
```
Comment (FranFin) [inline docs/quickstart.rst:45]: @jamoralp @pablogs9 
The original command was:
`cd /usr/local/bin && MicroXRCEAgent udp -p 2019 -r <references-file>`

File: docs/quickstart.rst [hunk 11]
```diff
@@ -2,335 +2,227 @@
...
+Running a Publisher-Subscriber example
+--------------------------------------
+
+In this section, we guide the user through the configuration and deployment of
+a simple publish/subscribe example where the communication is mediated by the *Agent* created above.
+
+Before considering the publisher and subscriber examples, it is useful to briefly summarize how the
+`Publisher` and `Subscriber` entities work, as well as to list the functions related to both entities.
+
+.. TODO: change text below according to the opportune entities.
```
Comment (FranFin) [inline docs/quickstart.rst:58]: @pablogs9

File: docs/quickstart.rst [hunk 12]
```diff
@@ -2,335 +2,227 @@
...
+    $ make
+
+At this point, it's possible to launch the :code:`RequestAdder` executable
+located in the folder :code:`Micro-XRCE-DDS-Client/build/examples/RequestAdder`, which'll make
+the *Client* send two integers as a request, and receive the sum of both integers as a response. ::
+
+    $ Micro-XRCE-DDS-Client/build/examples/RequestAdder/RequestAdder 127.0.0.1 2019
+
+.. TODO: decide which path to give to the user.
+.. TODO: check the command here.
```
Comment (FranFin) [inline docs/quickstart.rst:195]: @pablogs9

File: docs/quickstart.rst [hunk 13]
```diff
@@ -2,335 +2,227 @@
...
+^^^^^^^^^^^^^^^^^^^
+
+After having executed the Requester app, we can launch the :code:`ReplyAdder` executable,
+which is located in the folder :code:`Micro-XRCE-DDS-Client/build/examples/ReplyAdder`, which'll make
+this *Client* receive requests composed by two integers, sum both numbers, and finally send the response. ::
+
+    $ Micro-XRCE-DDS-Client/build/examples/ReplyAdder/ReplyAdder 127.0.0.1 2019
+
+.. TODO: decide which path to give to the user.
+.. TODO: check the command here.
```
Comment (FranFin) [inline docs/quickstart.rst:210]: @pablogs9

File: docs/p2p.rst (hunk 3 above)
Comment (pablogs9) [reply docs/p2p.rst:67]: If we have said `At this point,` I think that we should assume that user is in `build` folder, so the command should be:

```bash
 $ examples/PublishHelloWorld2P2/PublishHelloWorldClientP2P 127.0.0.1 2019
```

Change if you agree

File: docs/client.rst (hunk 8 above)
Comment (pablogs9) [reply docs/client.rst]: Addressed 99661293b331b86c3ab013c551050daeb6288026

File: docs/client_api.rst (hunk 9 above)
Comment (pablogs9) [reply docs/client_api.rst]: Adressed 814435f2b1e2eb00d7da475e3b8728c605e17f68

File: docs/getting_started.rst (hunk 2 above)
Comment (pablogs9) [reply docs/getting_started.rst:464]: I have check [the standard](https://www.omg.org/spec/DDS-XRCE/1.0/PDF) and there is no reference neither to `login` nor `log in`.

So if the concept of `log in` fits for you as "create a session in the agent", just accept this suggestion.

```suggestion
```

File: docs/quickstart.rst (hunk 10 above)
Comment (pablogs9) [reply docs/quickstart.rst:45]: If `-r <references-file>` is explained somewhere else I think that is not necessary to have it here in the quickstart.

If you agree, accept this suggestion:

```suggestion
```

File: docs/quickstart.rst (hunk 11 above)
Comment (pablogs9) [reply docs/quickstart.rst]: Addressed in e01cb842b09074e8409a97657409bcc63c475270

File: docs/quickstart.rst (hunk 12 above)
Comment (pablogs9) [reply docs/quickstart.rst:185]: If we have said `At this point,` I think that we should assume that user is in `build` folder, so the command should be:

```bash
 $ examples/RequestAdder/RequestAdder 127.0.0.1 2019
```

Change if you agree

File: docs/quickstart.rst (hunk 13 above)
Comment (pablogs9) [reply docs/quickstart.rst:200]: Same as in the others

File: docs/p2p.rst (hunk 3 above)
Comment (FranFin) [reply docs/p2p.rst:67]: agree.

**Comment by FranFin** [comment]
@pablogs9 TODO: write 2.0 release notes

**Pull request merged**